### DISCUSSION TITLE
does not support the latest version of Android

### DISCUSSION THREAD

**Discussion by skramovskiy** [airsdk/Adobe-Runtime-Support#3056] [General]
![image](https://github.com/airsdk/Adobe-Runtime-Support/assets/98939255/8732ce29-a4a2-4e0c-bf90-fd641fa6439c) Good afternoon. This error appears. They seem to have installed the latest versions of Air and Android Studio from the official website. But the error comes out - what to do? how do I check what's the matter?

**Comment by ajwfrost** [comment]
At a guess, the application has been packaged up with an older version of the AIR SDK and is no longer satisfying the rules that Android have for installations. You could open up the APK using Android Studio and check the manifest file, to see what the target API is... presumably this was side-loaded rather than downloaded from the Play Store?

Best thing really is to contact the developer and ask them to update/re-package it to satisfy the device/OS requirements.

thanks

**Reply by marchbold** [reply]
Check if you have overridden the air default by seeing if you have the following line in your app descriptor manifest additions:

```xml
<uses-sdk android:minSdkVersion="21" android:targetSdkVersion="33"/>
```

Make sure the `targetSdkVersion` is 33 or higher.

**Reply by skramovskiy** [reply]
Unfortunately, it didn’t help, I ordered both 33 and the last 34, what else could it be?

**Reply by FliplineStudios** [reply]
Did you download the correct Android SDK Build-Tools in Android Studio to match the targetSdkVersion you've set in the application descriptor XML?

**Reply by skramovskiy** [reply]
Yes! :(

**Reply by skramovskiy** [reply]
![image](https://github.com/airsdk/Adobe-Runtime-Support/assets/98939255/81d285f5-6fb8-4fd8-b843-7ac1f9507ac7)

**Reply by FliplineStudios** [reply]
Those look like the options under "SDK Platforms", I believe AIR also (or maybe only) uses the Build-Tools in the "SDK Tools" section.  I would show the package details for Android SDK Build-Tools and make sure you have 33.0.x installed (if you're targeting 33).

**Comment by skramovskiy** [comment]
trying to find a solution, I came across recommendations to check in build.gradle, but when publishing through animate, it is not used. Can you give advice on how else I can compile a publication for Android if I have a .swf or how else to find the cause of the error.

**Reply by ajwfrost** [reply]
Hi

We just had this reported elsewhere and their APK reproduced this issue on a device here .. but it turned out to be just because the APK had ARMv7 binary, and the device was ARMv8. It seems like in Android 14, they flag this warning if you're running a 32-bit version of native code...

In Animate, please make sure you select the "ARMv8" option in the "AIR for Android Settings" dialog box.

thanks

**Reply by skramovskiy** [reply]
Большое спасибо! Я убил столько времени пытаясь найти решение а все дело в одном селекторе! СПасибо огромное!
